pkg/controller/snapshot_create_test.go

```diff
@@ -0,0 +1,418 @@
+			expectedSnapshots: newSnapshotArray("snap6-3", validSecretClass, "snapcontent-snapuid6-3", "snapuid6-3", "claim6-3", false, nil, metaTimeNowUnix, getSize(defaultSize)),
+			initialClaims:     newClaimArray("claim6-3", "pvc-uid6-3", "1Gi", "volume6-3", v1.ClaimBound, &classEmpty),
+			initialVolumes:    newVolumeArray("volume6-3", "pv-uid6-3", "pv-handle6-3", "1Gi", "pvc-uid6-3", "claim6-3", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
+			initialSecrets:    []*v1.Secret{secret()},
+			expectedCreateCalls: []createCall{
+				{
+					snapshotName: "snapshot-snapuid6-3",
+					volume:       newVolume("volume6-3", "pv-uid6-3", "pv-handle6-3", "1Gi", "pvc-uid6-3", "claim6-3", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
+					parameters:   class5Parameters,
+					secrets:      map[string]string{"foo": "bar"},
+					// information to return
+					driverName: mockDriverName,
+					size:       defaultSize,
+					snapshotId: "sid6-3",
+					timestamp:  timeNow,
+					status: &csi.SnapshotStatus{
+						Type:    csi.SnapshotStatus_READY,
+						Details: "success",
+					},
+				},
+			},
+			errors: noerrors,
+			test:   testSyncSnapshot,
+		},
+		{
+			name:              "6-4 - successful create snapshot with snapshot class empty-secret-class",
+			initialContents:   nocontents,
+			expectedContents:  newContentArray("snapcontent-snapuid6-4", emptySecretClass, "sid6-4", "pv-uid6-4", "volume6-4", "snapuid6-4", "snap6-4", getSize(defaultSize), &timeNow),
+			initialSnapshots:  newSnapshotArray("snap6-4", emptySecretClass, "", "snapuid6-4", "claim6-4", false, nil, nil, nil),
+			expectedSnapshots: newSnapshotArray("snap6-4", emptySecretClass, "snapcontent-snapuid6-4", "snapuid6-4", "claim6-4", false, nil, metaTimeNowUnix, getSize(defaultSize)),
+			initialClaims:     newClaimArray("claim6-4", "pvc-uid6-4", "1Gi", "volume6-4", v1.ClaimBound, &classEmpty),
+			initialVolumes:    newVolumeArray("volume6-4", "pv-uid6-4", "pv-handle6-4", "1Gi", "pvc-uid6-4", "claim6-4", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
+			initialSecrets:    []*v1.Secret{emptySecret()},
+			expectedCreateCalls: []createCall{
+				{
+					snapshotName: "snapshot-snapuid6-4",
+					volume:       newVolume("volume6-4", "pv-uid6-4", "pv-handle6-4", "1Gi", "pvc-uid6-4", "claim6-4", v1.VolumeBound, v1.PersistentVolumeReclaimDelete, classEmpty),
+					parameters:   class4Parameters,
+					secrets:      map[string]string{},
+					// information to return
+					driverName: mockDriverName,
+					size:       defaultSize,
+					snapshotId: "sid6-4",
+					timestamp:  timeNow,
+					status: &csi.SnapshotStatus{
+						Type:    csi.SnapshotStatus_READY,
+						Details: "success",
+					},
+				},
+			},
+			errors: noerrors,
+			test:   testSyncSnapshot,
+		},
+		//{
+		//	name:                  "6-5 - successful create snapshot with default snapshot class",
```

Comment:
Why is this commented out? I don't see any other test for default snapshot class.

Reply:
When I add test, I found this test case failed due to this bug https://github.com/kubernetes-csi/external-snapshotter/pull/7#pullrequestreview-148480829,  I can uncomment it since the bug has been fixed.

Reply:
Ok, then please add a TODO comment to enable the test when the code is fixed.